### PR TITLE
[dart][ci] Move Dart samples from CircleCI to Github actions

### PR DESCRIPTION
**Pull request by kuhnroyal**
**Suggestions/Assumptions**
* future release branches (`5.2.x`) should be named `release/5.2.x` or some use different pattern that can be globbed in the workflow files
* there should be a separate workflow for Samples of each platform/language

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
@wing328 This also fixes a new formatting CI failure due to yesterdays Flutter/Dart release which now gets installed in CircleCI.

File: .github/workflows/samples-dart.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - release/**
+  pull_request:
+    branches:
+      - master
+      - release/**
```
Comment (wing328) [inline .github/workflows/samples-dart.yaml:11]: What about using something like [5-9].[0-9].[0-9]?

ref: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags

Comment (kuhnroyal) [reply .github/workflows/samples-dart.yaml:11]: This will work for `pull_request` but for `push` it will also match tags. I guess that could be ok.

Comment (kuhnroyal) [reply .github/workflows/samples-dart.yaml:11]: Nvm. tags are prefixed so this should work.

Comment (kuhnroyal) [reply .github/workflows/samples-dart.yaml:11]: But you always use `x` for the patch version?

Comment (wing328) [reply .github/workflows/samples-dart.yaml]: yup, please use `[5-9].[0-9].x`

**Comment by wing328** [comment]
Both drone.io, travis CI failures have been fixed in the master.

**Pull request merged**